src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java

```diff
@@ -90,28 +91,39 @@ public PropagatedContingency(Contingency contingency, int index, Set<String> bra
                 branchIdsToOpen.add(terminal.getConnectable().getId());
             }
             if (terminal.getConnectable() instanceof Generator) {
```

Comment:
@flo-dup I am not sure that the following lines are relevant and useful. Can you explain me why you introduce this loop on terminals to disconnect and if my code is relevant when we loose a load, a generator or a shunt ? Thanks a lot !

Reply:
Well, I didn't introduce that loop, but these lines are probably quite relevant and useful as they fill up the lists of disconnected equipments with the equipments traversed during propagation. I think those lists are then useful in the analysis - at least they are used a lot. But I feel like I misunderstood your question?

About the code correctness, I don't really get what you want me to check? indeed, you didn't really change that loop

Reply:
A few things after trying to understand this code though:
- I don't get why you store the targetP of the generator? it seems that it's not used in the analysis, you only use the id
- The Triple object you store for loads makes the code unclear, why didn't you introduce a specific object instead?
- Do we need to store parameters of disconnected shuntCompensators with this Pair list? whereas it looks like the LfShuntCompensator already holds that info
- and strangely `getShuntIdsToLoose` is not called in the SensitivityAnalysis, unlike all other lists, so there's maybe something missing? Not current PR's topic though.

Reply:
It is just the unit tests never go into that lines except the first if statement. So... I was wondering if it was relevant as it was introduced by non graph experts!

Reply:
> * Do we need to store parameters of disconnected shuntCompensators with this Pair list? whereas it looks like the LfShuntCompensator already holds that info

Shunts are aggregated by buses in the LF network

Reply:
> * and strangely `getShuntIdsToLoose` is not called in the SensitivityAnalysis, unlike all other lists, so there's maybe something missing? Not current PR's topic though.

Shunt contingencies don't work at ALL in security and sensitivity analysis

Reply:
> * I don't get why you store the targetP of the generator? it seems that it's not used in the analysis, you only use the id

Right, removed